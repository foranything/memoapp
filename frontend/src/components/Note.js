import React, {Component} from 'react';
import "./Note.css"

class Note extends Component {
  render() {
    const {username, title, content} = this.props;
    return (
      <div className='Note col s12 m4 l3'>
        <div className='DeleteBtn'>
        </div>
        <div className='card yellow lighten-4'>
          <div className='card-content black-text'>
            <span className='card-title'>{title}</span>
            <p>{content}</p>
            <p>{username}</p>
          </div>
        </div>
      </div>
    );
  }
}

export default Note;

// {/*<div>*/}
// {/*Note*/}
// {/*<span> {index}</span>*/}
// {/*<span> {username} </span>*/}
// {/*<span> {title} </span>*/}
// {/*<span> {content} </span>*/}
// {/*</div>*/}