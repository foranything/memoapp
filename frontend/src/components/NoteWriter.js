import React, {Component} from 'react';

class NoteWriter extends Component {
  state = {
    title:'',
    content:'',
    password:'',
  };
  
  handleChange=(e)=>{
    this.setState({
        [e.target.name]:e.target.value,
      })
  };
  //
  // handleClick=()=>{
  //   const {save} = this.props;
  //   save(this.state);
  //   this.setState({
  //     title:'',
  //     content:'',
  //     password:'',
  //   })
  // };
  
  handleSubmit=(e)=>{
    const {save} = this.props;
    save(this.state);
    this.setState({
      title:'',
      content:'',
      password:'',
    })
    e.preventDefault();
  };
  
  render() {
    // console.log(this.state);
    const {title, content, password} = this.state;
    const {handleChange, handleSubmit} = this;
    
    return (
      <form className='NoteWriter' onSubmit={handleSubmit}>
        <input
          placeholder='제목을 입력하세요.'
          name='title'
          value={title}
          onChange={handleChange}
        />
        <input
          placeholder='내용을 입력하세요.'
          name='content'
          value={content}
          onChange={handleChange}
        />
        <input
          placeholder='비밀번호를 입력하세요.'
          type='password'
          name='password'
          value={password}
          onChange={handleChange}
        />
        <button type='submit'>
          등록
        </button>
      </form>
    );
  }
}
export default NoteWriter;