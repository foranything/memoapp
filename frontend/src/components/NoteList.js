import React, { Component } from 'react';
import Note from './Note'

class NoteList extends Component {
  render() {
    const { notes } = this.props;
    return (
      <div className='NoteList'>
        {notes.map((note, index) =>
          <Note username={note.username} title={note.title} content={note.content} key={index} />)}
      </div>
    );
  }
}

export default NoteList;

// {/*{notes.map(()=>(<Note/>))}*/}
// {/*notes.map(note=><Note note={note}/>)*/}