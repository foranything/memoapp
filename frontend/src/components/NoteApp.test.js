import React from 'react';
import NoteWriter from './NoteWriter';
import NoteList from './NoteList';
import Note from './Note';
import { render, fireEvent } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";


describe('<Note />', () => {
  it('renders Note', () => {
    const utils = render(<Note index={1} username='김대운' title='제목' content='내용'/>);
    utils.getByText('김대운'); // Note 존재유무 확인
    expect(utils.container).toMatchSnapshot();
    utils.getByText('김대운'); // 김대운 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('제목'); // 제목 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('내용'); // 내용 라는 텍스트를 가진 엘리먼트가 있는지 확인
  
  });
});

describe('<NoteList />', () => {
  const sampleNotes = [
    {id: 1, username:'user1', title: 'title1', content: 'content1'},
    {id: 2, username:'user2', title: 'title2', content: 'content2'},
    {id: 3, username:'user3', title: 'title3', content: 'content3'},
  ]
  it('renders NoteList', () => {
    const utils = render(<NoteList notes={sampleNotes}/>);
    expect(utils.container).toMatchSnapshot();
    utils.getByText('user1'); // 김대운 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('title1'); // 제목 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('content1'); // 내용 라는 텍스트를 가진 엘리먼트가 있는지 확인
    // expect(utils.component.contains(<Note />)).toBe(true);
  });
});

describe('<NoteWriter />', () => {
  const save = jest.fn();
  it('renders NoteWriter', () => {
    
    // (1) UI 구성하기
    const utils = render(<NoteWriter save={save}/>);
    expect(utils.container).toMatchSnapshot();
    const input1 = utils.getByPlaceholderText('제목을 입력하세요.');
    const input2 = utils.getByPlaceholderText('내용을 입력하세요.');
    const input3 = utils.getByPlaceholderText('비밀번호를 입력하세요.');
    const btn = utils.getByText('등록');
    
    // (2) Input 상태 관리하기
    // 수정하고
    fireEvent.change(input1, {
      target: {
        value: '여기는제목'
      }
    });
    fireEvent.change(input2, {
      target: {
        value: '여기는내용'
      }
    });
    fireEvent.change(input3, {
      target: {
        value: '여기는비밀번호'
      }
    });
    expect(input1).toHaveAttribute('value', '여기는제목')
    expect(input2).toHaveAttribute('value', '여기는내용')
    expect(input3).toHaveAttribute('value', '여기는비밀번호')
    
    // (3) Submit 이벤트 관리하기
    // 버튼 클릭해서 리턴값 확인하기
    fireEvent.click(btn);
    expect(save).toBeCalledWith(
      {
        title:'여기는제목',
        content:'여기는내용',
        password:'여기는비밀번호'
      });
    expect(input1).toHaveAttribute('value', ''); // input1이 비워져야함
    expect(input2).toHaveAttribute('value', ''); // input2이 비워져야함
    expect(input3).toHaveAttribute('value', ''); // input3이 비워져야함
  });
});


// describe('<NoteWriter />', () => {
//   it('renders NoteList', () => {
//     const utils = render(<NoteWriter />);
//     expect(utils.container).toMatchSnapshot();
//     utils.getByText('user1'); // 김대운 라는 텍스트를 가진 엘리먼트가 있는지 확인
//     utils.getByText('title1'); // 제목 라는 텍스트를 가진 엘리먼트가 있는지 확인
//     utils.getByText('content1'); // 내용 라는 텍스트를 가진 엘리먼트가 있는지 확인
//   });
// });



//
// it('creates new todo', () => {
//   const { getByPlaceholderText, getByText } = render(<TodoApp />);
//   // 이벤트를 발생시켜서 새 항목을 추가하면
//   fireEvent.change(getByPlaceholderText('할 일을 입력하세요'), {
//     target: {
//       value: '새 항목 추가하기'
//     }
//   });
//   fireEvent.click(getByText('등록'));
//   // 해당 항목이 보여져야합니다.
//   getByText('새 항목 추가하기');
// });
//
// it('toggles todo', () => {
//   const { getByText } = render(<TodoApp />);
//   // TDD 배우기 항목에 클릭 이벤트를 발생시키고 text-decoration 속성이 설정되는지 확인
//   const todoText = getByText('TDD 배우기');
//   expect(todoText).toHaveStyle('text-decoration: line-through;');
//   fireEvent.click(todoText);
//   expect(todoText).not.toHaveStyle('text-decoration: line-through;');
//   fireEvent.click(todoText);
//   expect(todoText).toHaveStyle('text-decoration: line-through;');
// });
//
// it('removes todo', () => {
//   const { getByText } = render(<TodoApp />);
//   const todoText = getByText('TDD 배우기');
//   const removeButton = todoText.nextSibling;
//   fireEvent.click(removeButton);
//   expect(todoText).not.toBeInTheDocument(); // 페이지에서 사라졌음을 의미함
// });