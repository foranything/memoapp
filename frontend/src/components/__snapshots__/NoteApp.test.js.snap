// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Note /> renders Note 1`] = `
<div>
  <div
    class="Note col s12 m4 l3"
  >
    <div
      class="DeleteBtn"
    />
    <div
      class="card yellow lighten-4"
    >
      <div
        class="card-content black-text"
      >
        <span
          class="card-title"
        >
          제목
        </span>
        <p>
          내용
        </p>
        <p>
          김대운
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NoteList /> renders NoteList 1`] = `
<div>
  <div
    class="NoteList"
  >
    <div
      class="Note col s12 m4 l3"
    >
      <div
        class="DeleteBtn"
      />
      <div
        class="card yellow lighten-4"
      >
        <div
          class="card-content black-text"
        >
          <span
            class="card-title"
          >
            title3
          </span>
          <p>
            content3
          </p>
          <p>
            user3
          </p>
        </div>
      </div>
    </div>
    <div
      class="Note col s12 m4 l3"
    >
      <div
        class="DeleteBtn"
      />
      <div
        class="card yellow lighten-4"
      >
        <div
          class="card-content black-text"
        >
          <span
            class="card-title"
          >
            title2
          </span>
          <p>
            content2
          </p>
          <p>
            user2
          </p>
        </div>
      </div>
    </div>
    <div
      class="Note col s12 m4 l3"
    >
      <div
        class="DeleteBtn"
      />
      <div
        class="card yellow lighten-4"
      >
        <div
          class="card-content black-text"
        >
          <span
            class="card-title"
          >
            title1
          </span>
          <p>
            content1
          </p>
          <p>
            user1
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NoteWriter /> renders NoteWriter 1`] = `
<div>
  <form
    class="NoteWriter"
  >
    <input
      name="title"
      placeholder="제목을 입력하세요."
      value=""
    />
    <input
      name="content"
      placeholder="내용을 입력하세요."
      value=""
    />
    <input
      name="password"
      placeholder="비밀번호를 입력하세요."
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      등록
    </button>
  </form>
</div>
`;
